app/src/reducers/vesselInfo.js

```diff
@@ -12,15 +12,21 @@ import {
   SET_PINNED_VESSEL_HUE,
   LOAD_PINNED_VESSEL,
   SET_PINNED_VESSEL_TITLE,
-  TOGGLE_PINNED_VESSEL_EDIT_MODE
+  TOGGLE_PINNED_VESSEL_EDIT_MODE,
+  SET_RECENT_VESSELS_VISIBILITY,
+  SET_RECENT_VESSEL_HISTORY
 } from 'actions';
 import { DEFAULT_TRACK_HUE } from 'constants';
 
 const initialState = {
   tracks: [],
   details: [],
   detailsStatus: null,
-  pinnedVesselEditMode: false
+  pinnedVesselEditMode: false,
+  history: [],
+  recentVesselModal: {
```

Comment:
I would move this to the map reducer. I don't like all modals being there, but we either keep them there, or  refactor *all* of them into a new reducer.